v4/examples/mistral-vllm.mdx

```diff
@@ -0,0 +1,157 @@
+---
+title: "Mistral 7B with vLLM"
+description: "Deploy Mistral 7B with vLLM"
+---
+
+In this tutorial, we will show you how to deploy Mistral 7B using the popular vLLM inference framework.
+
+To see the final implementation, you can view it [here](https://github.com/CerebriumAI/examples/tree/master/4-faster-inference-with-vllm)
+
+## Basic Setup
+
+It is important to think of the way you develop models using Cerebrium should be identical to developing on a virtual machine or Google Colab - so converting this should be very easy!
+Please make sure you have the Cerebrium package installed and have logged in. If not, please take a look at our docs [here](https://docs.cerebrium.ai/cerebrium/getting-started/installation)
+
+First we create our project:
+
+```
+cerebrium init mistral-vllm
+```
+
+We need certain Python packages to implement this project. Lets add those to our **[cerebrium.dependencies.pip]** in our `cerebrium.toml` file:
+
+```
+[cerebrium.dependencies.pip]
+sentencepiece = "latest"
+torch = ">=2.0.0"
+vllm = "latest"
+transformers = ">=4.35.0"
+accelerate = "latest"
+xformers = "latest"
+```
+
+Our **main.py** file will contain our main Python code. This is a relatively simple implementation, so we can do everything in 1 file. We would like a user to send in a link to a YouTube video with a question and return to them the answer as well as the time segment of where we got that response.
+So let us define our request object.
+
+```python
+from pydantic import BaseModel
+
+class Item(BaseModel):
+    prompt: str
+    temperature: Optional[float] = 0.8
+    top_p: Optional[float] = 0.75
+    top_k: Optional[float] = 40
+    max_tokens: Optional[int] = 256
+    frequency_penalty: Optional[float] = 1
+```
+
+Above, we use Pydantic as our data validation library. We specify the parameters that are required as well as the parameters that are not (ie: using the Optional keyword) as well as assign defaults to some values. Prompt is the only required parameter so if it is not present in the request, the user will automatically receive an error message.
+
+## vLLM Implementation
+
+Below, we will use the Whisper model from OpenAI to convert the video audio to text. We will then split the text into its phrase segments with its respective timings, so we know the exact source of where our model got the answer from.
+
+```python
+import torch
+from vllm import LLM, SamplingParams
+
+llm = LLM(model="mistralai/Mistral-7B-Instruct-v0.1", dtype="bfloat16")
+
+def predict(prompt, temperature, top_p, top_k, max_tokens, frequency_penalty, run_id):
```

Comment:
These are positional arguments so all of them become required, which is in conflict with the pydantic model. You'll get an error when trying to call the endpoint if you don't provide them. Structure of params needs to change (To contain defaults) and so does the pydantic model (to not include defaults)